tools/src/test/java/org/apache/kafka/tools/TopicCommandTest.java

```diff
@@ -0,0 +1,283 @@
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.verify;
+
+@Timeout(value = 60)
+public class TopicCommandTest {
+    private String bootstrapServer = "localhost:9092";
+    private String topicName = "topicName";
+
+    @Test
+    public void testIsNotUnderReplicatedWhenAdding() {
+        List<Integer> replicaIds = Arrays.asList(1, 2);
+        List<Node> replicas = new ArrayList<>();
+        for (int id : replicaIds) {
+            replicas.add(new Node(id, "localhost", 9090 + id));
+        }
+
+        TopicCommand.PartitionDescription partitionDescription = new TopicCommand.PartitionDescription("test-topic",
+            new TopicPartitionInfo(0, new Node(1, "localhost", 9091), replicas,
+                Collections.singletonList(new Node(1, "localhost", 9091))),
+            null, false,
+                new PartitionReassignment(replicaIds, Arrays.asList(2), Collections.emptyList())
+        );
+
+        assertFalse(partitionDescription.isUnderReplicated());
+    }
+
+    @Test
+    public void testAlterWithUnspecifiedPartitionCount() {
+        String[] options = new String[] {" --bootstrap-server", bootstrapServer, "--alter", "--topic", topicName};
+        assertInitializeInvalidOptionsExitCode(1, options);
+    }
+
+    @Test
+    public void testConfigOptWithBootstrapServers() {
+        assertInitializeInvalidOptionsExitCode(1,
+            new String[] {"--bootstrap-server", bootstrapServer, "--alter", "--topic", topicName,
+                "--partitions", "3", "--config", "cleanup.policy=compact"});
+        assertInitializeInvalidOptionsExitCode(1,
+            new String[] {"--bootstrap-server", bootstrapServer, "--alter", "--topic", topicName,
+                "--partitions", "3", "--delete-config", "cleanup.policy"});
+        TopicCommand.TopicCommandOptions opts =
+            new TopicCommand.TopicCommandOptions(
+                new String[] {"--bootstrap-server", bootstrapServer, "--create", "--topic", topicName, "--partitions", "3",
+                    "--replication-factor", "3", "--config", "cleanup.policy=compact"});
+        assertTrue(opts.hasCreateOption());
+        assertEquals(bootstrapServer, opts.bootstrapServer().get());
+        assertEquals("cleanup.policy=compact", opts.topicConfig().get().get(0));
+    }
+
+    @Test
+    public void testCreateWithPartitionCountWithoutReplicationFactorShouldSucceed() {
+        TopicCommand.TopicCommandOptions opts = new TopicCommand.TopicCommandOptions(
+            new String[] {"--bootstrap-server", bootstrapServer,
+                "--create",
+                "--partitions", "2",
+                "--topic", topicName});
+        assertTrue(opts.hasCreateOption());
+        assertEquals(topicName, opts.topic().get());
+        assertEquals(2, opts.partitions().get());
```

Comment:
The original test called `opts.checkArgs()`, should we do that here to?

Reply:
`checkArgs()` is called in the constructor of  `TopicCommandOptions`. It is redundant to call  `opts.checkArgs`